src/core/geometry/qgsgeometryutils.cpp

```diff
@@ -1173,6 +1174,131 @@ double QgsGeometryUtils::averageAngle( double a1, double a2 )
   return normalizedAngle( resultAngle );
 }
 
+double QgsGeometryUtils::skewLinesDistance( const QVector3D &P1, const QVector3D &P12,
```

Comment:
I'm unsure whether QVector3D is appropriate here. It uses floats for x/y/z, but I think that this method really should be using doubles.

Reply:
You're right. I used QVector3D since it is already used for 3D. I can create a new QgsVector3D class using doubles if you prefer.

Reply:
Sorry... I hadn't seen that QgsVector3D class exist since I started working on my algorithm...